velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -284,8 +286,20 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
       if (isFileColumnNamesReadAsLowerCase()) {
         folly::toLowerAscii(childName);
       }
-      auto childRequestedType =
-          requestedType ? requestedType->asRow().findChild(childName) : nullptr;
+
+      TypePtr childRequestedType = nullptr;
+      if (requestedType) {
+        if (requestedType->isRow()) {
+          auto fileTypeIdx =
+              requestedType->asRow().getChildIdxIfExists(childName);
```

Comment:
There might be some conflicts between this change and #10517.  We may need to wait that one to be merged first.

Reply:
HI @Yuhta , We need to get this merged first. Since https://github.com/facebookincubator/velox/pull/10517 has dependency on this PR since this PR is also taking care of fixing https://github.com/facebookincubator/velox/issues/10680#issuecomment-2286249363
Once this is merged, I can rebase my PR as well.